raiwidgets/tests/test_cohort.py

```diff
@@ -497,40 +497,46 @@ def test_cohort_serialization_single_value_method(self, method):
                                        arg=[65], column='age')
         cohort_1 = Cohort(name="Cohort New")
         cohort_1.add_cohort_filter(cohort_filter_1)
-        json_str = json.dumps(cohort_1,
-                              default=cohort_filter_json_converter)
+        json_str = cohort_1.to_json()
 
         assert 'Cohort New' in json_str
         assert method in json_str
         assert '[65]' in json_str
         assert 'age' in json_str
 
-    def test_cohort_serialization_in_range_method(self):
+    def test_cohort_serialization_deserialization_in_range_method(self):
         cohort_filter_1 = CohortFilter(
             method=CohortFilterMethods.METHOD_RANGE,
             arg=[65.0, 70.0], column='age')
         cohort_1 = Cohort(name="Cohort New")
         cohort_1.add_cohort_filter(cohort_filter_1)
-        json_str = json.dumps(cohort_1,
-                              default=cohort_filter_json_converter)
 
+        json_str = cohort_1.to_json()
         assert 'Cohort New' in json_str
         assert CohortFilterMethods.METHOD_RANGE in json_str
         assert '65.0' in json_str
         assert '70.0' in json_str
         assert 'age' in json_str
 
+        cohort_1_new = Cohort.from_json(json_str)
+        assert cohort_1_new.name == cohort_1.name
+        assert len(cohort_1_new.cohort_filter_list) == \
+            len(cohort_1.cohort_filter_list)
+        assert cohort_1_new.cohort_filter_list[0].method == \
+            cohort_1.cohort_filter_list[0].method
```

Comment:
Should there be a test case that asserts that the data types remain consistent if you do to_json AND from_json? Wouldn't want a number to become string in the process, or something like that.

Reply:
Added an `__eq__` is `Cohort` class that returns True in case the cohort and its underlying filters are equal when we serialize using to_json and deserialize using from_json.

Reply:
That makes sure that the types are respected across serialization and de-serialization.